testcases/GcovSetup.py

```diff
@@ -0,0 +1,164 @@
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+'''
+gcov
+----
+
+A "Test Case" for extracting GCOV code coverage data for linux kernel.
+Installs the kernel source and build the kernel with gcov enabled.
+'''
+
+import os
+import tempfile
+import unittest
+import time
+import OpTestConfiguration
+from common.OpTestSystem import OpSystemState
+from common.OpTestUtil import OpTestUtil
+
+import logging
+import OpTestLogger
+log = OpTestLogger.optest_logger_glob.get_logger(__name__)
+
+
+class GcovBuild(unittest.TestCase):
+    '''
+    Install the distro src  kernel and gcov is enabled and booted with
+    newly installed kernel
+    '''
+    def setUp(self):
+        self.conf = OpTestConfiguration.conf
+        self.host_cmd_timeout = self.conf.args.host_cmd_timeout
+        self.cv_HOST = self.conf.host()
+        self.cv_SYSTEM = self.conf.system()
+        self.util = OpTestUtil(OpTestConfiguration.conf)
+
+
+    def runTest(self):
+        '''
+        Running the gcov test
+        '''
+        self.distro_name = self.util.distro_name()
+        log.info("OS: %s" %self.distro_name)
+        if self.distro_name == 'rhel':
+            self.installer = "yum install"
+        elif self.distro_name == 'sles':
+            self.installer = "zypper install"
+        dep_packages = ["rpm-build", "yum-utils", "gcc*", "perl*", "tiny*"]
+        log.info(f"\nInstalling following dependency packages\n {dep_packages}")
+        for pkg in dep_packages:
+            self.cv_HOST.host_run_command(f"{self.installer} {pkg} -y")
+        log.info("\nInstalling the ditro src...")
+        if self.distro_name == 'rhel':
+            src_path = self.util.get_distro_src('kernel', '/root', "-bp")
+        elif self.distro_name == 'sles':
+            src_path = self.util.get_distro_src('kernel-default', '/root', "-bp")
+        out = self.cv_HOST.host_run_command(f"ls {src_path}")
+        for line in out:
+            if line.startswith("linux-"):
+                src_path = os.path.join(src_path, line)
+                break
+        log.info("\n\nsource path = %s" %src_path)
+        log.info("\nadding gcov_param....")
+        self.kernel_config(src_path)
+        log.info("Building the new kernel...")
+        self.build_and_boot(src_path)
+        log.info("rebooting the lpar after building the src with kernel parameters")
+        self.cv_SYSTEM.goto_state(OpSystemState.OFF)
+        self.cv_SYSTEM.goto_state(OpSystemState.OS)
```

Comment:
can we check if Gcov path or kernel is there

Reply:
done